zmon_worker_monitor/zmon_worker/tasks/main.py

```diff
@@ -1042,6 +1051,25 @@ def _store_check_result(self, req, result):
         self.con.lpush(key, value)
         self.con.ltrim(key, 0, DEFAULT_CHECK_RESULTS_HISTORY_LENGTH - 1)
 
+    def _check_result_limit(self, result):
+        if isinstance(result, dict):
+            key_count = len(flatten(result).keys())
+            if key_count > self.max_result_keys:
+                raise ResultSizeError(
+                    'Result keys count ({}) exceeded the maximum value: {}'.format(key_count, self.max_result_keys))
+
+        result_str = ''
+        if isinstance(result, dict) or isinstance(result, list):
```

Comment:
let's keep it simple and use `json.dumps(...)` always (one less branch to maintain)